ethdb/leveldb/leveldb.go

```diff
@@ -360,7 +355,11 @@ func (db *Database) meter(refresh time.Duration) {
 		db.nonlevel0CompGauge.Update(int64(stats.NonLevel0Comp))
 		db.seekCompGauge.Update(int64(stats.SeekComp))
 
-		// update tables amount
+		// goleveldb may have more than 7 layers, so we'll only append the missing layers
+		for i := len(db.levelsGauge); i < len(stats.LevelTablesCounts); i++ {
+			db.levelsGauge = append(db.levelsGauge, metrics.NewRegisteredGauge(namespace+fmt.Sprintf("tables/level%v", i), nil))
+		}
```

Comment:
Since you removed the initiation from `NewCustom`, I don't see why you need to append here. Right below this clause, there's this loop: 
```golang
		for i, tables := range stats.LevelTablesCounts {
			db.levelsGauge[i].Update(int64(tables))
		}
```
Couldn't we just do the (complete) initiation of gauges there? 
Am I missing something?

Reply:
Ah, indeed, I was missing something. So this method will run over and over again, and this snippet is also in a loop internally. Yeah ignore what I said

Reply:
Still, can be combined: 
```golang
		for i, tables := range stats.LevelTablesCounts {
			// Append metrics for additional layers
			if i >= len(stats.LevelTablesCounts) {
				db.levelsGauge = append(db.levelsGauge, metrics.NewRegisteredGauge(namespace+fmt.Sprintf("tables/level%v", i), nil))
			}
			db.levelsGauge[i].Update(int64(tables))
		}
```

Reply:
s/len(stats.LevelTablesCounts)/len(db.levelsGauge)/

Reply:
@holiman updated, ptal